### PR TITLE
stm32wl55 internal (die) temperature sensor 

### LINKED ISSUE #49897: STM32: NUCLEO_WL55JC internal (die) temperature incorrect
**Describe the bug**
I am facing an issue with stock Zephyr (latest main), unmodified, and reading internal die temperature from the ADC. I am using the sample provided in samples/sensor/stm32_temp_sensor and the following overlay:

```
&die_temp {
	compatible = "st,stm32-temp-cal";
	status = "okay";
	ts-cal1-addr = <0x1FFF75A8>;
	ts-cal2-addr = <0x1FFF75C8>;
	ts-cal1-temp = <30>;
	ts-cal2-temp = <130>;
	ts-cal-vrefanalog = <3300>;
	io-channels = <&adc1 12>;
};

&adc1 {
	status = "okay";
	has-temp-channel;
	has-vref-channel;
	has-vbat-channel;
	vref-mv = <3300>;
};
```

To the best of my knowledge (and many double-checks with the datasheet) I believe the register addresses and calibration temperatures to be correct.

On running the sample I get:

```
*** Booting Zephyr OS build zephyr-v3.1.0-4713-g4aa408484013  ***
STM32 Die temperature sensor test
Current temperature: -121.5 °C
Current temperature: -121.5 °C
Current temperature: -121.8 °C
Current temperature: -121.1 °C
Current temperature: -121.1 °C
Current temperature: -120.5 °C
```

**To Reproduce**
1. Copy the dts things in to samples/sensor/stm32_temp_sensor/boards/nucleo_wl55jc.overlay
2. west build --pristine -b nucleo_wl55jc zephyr/samples/sensor/stm32_temp_sensor/
3. west flash

**Expected behavior**
Rational temperature readings somewhere in the room-temperature range.

**Impact**
Just a bit of an inconvenience for LoRaWAN transmissions.  As a side note, the battery reading is perfectly OK.

**Logs and console output**
See description.

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain Zephry SDK 0.15.0
 - Latest main

### PR DESCRIPTION
**Pull request by FRASTM**
Configures the ADC channel and common registers to give a correct die-temperature sensor value

The ADC of the stm32wl should not enter sleep mode during conversion, like other mcus but LL function differs.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/49897

Thanks to this PR the die temperature on the nucleo_wl55jc platform with samples/sensor/stm32_temp_sensor/ is measured at 
`Current temperature: 20.0  °C`

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Comment by FRASTM** [comment]
simplify w/o overlay file for the samples/sensor/stm32_temp_sensor/ on  nucleo_wl55 board

File: drivers/adc/adc_stm32.c [hunk 1]
```diff
@@ -1129,6 +1132,13 @@ static int adc_stm32_init(const struct device *dev)
 	 */
 
 	LL_ADC_DisableDeepPowerDown(adc);
+#elif defined(CONFIG_SOC_SERIES_STM32WLX)
+	/*
+	 * WL series STM32 needs to be awaken from deep sleep
+	 * mode, and restore its calibration parameters if there are some
+	 * previously stored calibration parameters.
+	 */
```
Comment (erwango) [inline drivers/adc/adc_stm32.c:1140]: "The ADC clock must be disabled by clock gating during CPU1 sleep/stop"

**Review by erwango — CHANGES_REQUESTED** [review]
One comment otherwise LGTM

**Comment by FRASTM** [comment]
Rebase on 55c4e06b35639d525d214cb3432740a489b7147f
Changing comment

**Review by erwango — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**